### PR TITLE
Re-add erroneously removed lcfiplus patch

### PR DESCRIPTION
**Pull request by vvolkl**
BEGINRELEASENOTES
- Re-add erroneously removed lcfiplus patch

ENDRELEASENOTES

### CODE REVIEW

File: packages/lcfiplus/package.py [hunk 1]
```diff
@@ -20,6 +20,10 @@ class Lcfiplus(CMakePackage, Ilcsoftpackage):
     version('0.10.1', sha256='4eac91718b29de926f7cd5bc7aa879d157bfec8f4306ccd1d74785813569fde0')
     version('0.10',       sha256='0d4d27cd0d9407cd2f13e5a978be8c9389bc86c78c2eefd0ae7c060c4b7196c3')
 
+    patch('dict_0.10.1.patch', when='@0.10.1:')
+    patch('dict.patch', when='@0.10')
```
Comment (tmadlener) [inline packages/lcfiplus/package.py:24]:
```suggestion
    patch('dict.patch', when='@0.10')
```
The patch seems to be already there, only applying it has been removed.

This is what I get with the current version of the PR:
```console
==> Installing lcfiplus-0.10.1-avrsh3dcasqscrn27s26m332vrmn22i3
==> No binary for lcfiplus-0.10.1-avrsh3dcasqscrn27s26m332vrmn22i3 found: installing from source
==> Using cached archive: /resources/spack/var/spack/cache/_source-cache/archive/4e/4eac91718b29de926f7cd5bc7aa879d157bfec8f4306ccd1d74785813569fde0.tar.gz
==> Applied patch /resources/key4hep-spack/packages/lcfiplus/dict_0.10.1.patch
Reversed (or previously applied) patch detected!  Assume -R? [n]
```

**Comment by tmadlener** [comment]
Applying the patch fixes the problems I observed and the tests run smoothly now.

**Comment by tmadlener** [comment]
accidentally closed this

File: packages/lcfiplus/package.py (hunk 1 above)
Comment (tmadlener) [reply packages/lcfiplus/package.py:24]: Actually it should be `@0.10:` (i.e. open ended range)
```suggestion
    patch('dict.patch', when='@0.10:')
```

Comment (vvolkl) [reply packages/lcfiplus/package.py]: Thanks, it turns out `dict.patch` can indeed be applied to both `0.10` and `0.10.1`.

This shows again the pitfalls of using both two field and three field versions with spack: To only apply this to 0.10, one has to use the strange construct '@0.10:0.10.0'

**Pull request merged**